### PR TITLE
mgr/dashboard: Object gateway sync status cards keeps loading when multisite is not configured

### PR DESCRIPTION
**Pull request by aaSharma14**
Fixes: https://tracker.ceph.com/issues/62665

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
> we are still able to create zonegroups

@aaSharma14 do we have any other info that we can rely on to say multisite is configured? There should be something in the `radosgw-admin sync status` right?

**Comment by aaSharma14** [comment]
> > we are still able to create zonegroups
> 
> @aaSharma14 do we have any other info that we can rely on to say multisite is configured? There should be something in the `radosgw-admin sync status` right?

@nizamial09 ,In the scenario that @Pegonzal mentioned..the sync status command returns an error with exit code != 0 which means the command didn't even execute properly and we get the dashboard exception in that case..I have updated the PR to fix this

**Comment by aaSharma14** [comment]
> Hey Aashish some screenshots or display of the changes would be nice in the PR description,
> 
> as far I've seen since we are still able to create zonegroups even if multisite is not configured we can still end up encountering the issue, an example:
> screen-capture.28.webm
> 
> Also maybe we could add margins to the aler-panel for consistency within the dashboard image

Thanks @Pegonzal for testing out the scenario..I have updated the PR accordingly

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts [hunk 1]
```diff
@@ -118,20 +117,22 @@ export class RgwOverviewDashboardComponent implements OnInit, OnDestroy {
...
+              this.replicaZonesInfo = data['dataSyncInfo'];
+              this.metadataSyncInfo = data['metadataSyncInfo'];
+              if (this.replicaZonesInfo.length === 0) {
+                this.showMultisiteCard = false;
+                this.syncCardLoading = false;
+              }
+              [this.realm, this.zonegroup, this.zone] = data['primaryZoneData'];
+            }
           })
         )
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts:134]:
```suggestion
        ),
        catchError((error) => doThisWhenErrorHappens 
```

**Comment by nizamial09** [comment]
> didn't even execute properly and we get the dashboard exception in that case

if error happens you shouldn't return a data. Instead raising exception is the right way. You just need to properly handle it in the frontend by catching the error

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts:134]: for eg: https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts#L153

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.html [hunk 2]
```diff
@@ -112,11 +112,13 @@ <h1>{{ averageObjectSize | dimlessBinary}}</h1>
...
              i18n-title>
       <ng-template #notConfigured>
-        <cd-alert-panel type="info"
-                        i18n>
-          Multisite needs to be configured in order to see the multisite sync status.
-          Please consult the <cd-doc section="multisite"></cd-doc> on how to configure and enable the multisite functionality.
-        </cd-alert-panel>
+        <span class="pe-5 ps-5"> 
+          <cd-alert-panel type="info"
+                          i18n>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.html:117]: alert-panel has a spacing class
```suggestion
          <cd-alert-panel type="info"  
                          spacingClass='ps-5 pe-5'
                          i18n>
```

Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.html:117]: Didnt work in this case

**Review by nizamial09 — APPROVED** [review]
LGTM, Thank you!

**Comment by nizamial09** [comment]
@aaSharma14, make-check failure
```
[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 205 files
[lint:html    ] All files pass linting
[lint:prettier] src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] 

Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.
```

**Comment by aaSharma14** [comment]
jenkins make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Review by Pegonzal — APPROVED** [review]

**Pull request merged**